### PR TITLE
ci: Optimize CI processes

### PR DESCRIPTION
**Pull request by sy-records**
close #2452
close #2453

### CODE REVIEW

**Comment by Koooooo-7** [comment]
Could you don't remove the `update-emoji.yaml` and update it instead?
Then we can manually check the workflow in PR branches in Action since it already in `develop`.

**Comment by sy-records** [comment]
you can see https://github.com/sy-records/docsify/actions/runs/9442492589/job/26004732688

**Comment by sy-records** [comment]
You need to fork and test in your own repo, add a push event, remove the `if: github.repository == 'docsifyjs/docsify' `, and modify reviewer to yourself.

Please test in your own repo since testing that will initiate an invalid PR.

**Comment by Koooooo-7** [comment]
> You need to fork and test in your own repo, add a push event, remove the `if: github.repository == 'docsifyjs/docsify' `, and modify reviewer to yourself.

I see the `if: github.repository == 'docsifyjs/docsify' `  can prevent trigger on forks.
What I mean is we can just test here as a real situation.

> Please test in your own repo since testing that will initiate an invalid PR.

I think raise an invalid PR is fine to me since we won't  merge it.

**Comment by Koooooo-7** [comment]
Plz take a look on https://github.com/docsifyjs/docsify/actions/runs/9447704932/job/26020091254 .

File: .github/workflows/emoji.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          create_branch: true
+          file_pattern: 'src/core/render/emoji-data.js docs/emoji.md'
+          push_options: '--force'
+
+      - name: Create Pull Request
+        if: ${{ steps.auto-commit-action.outputs.changes_detected == 'true' }}
+        run: |
+          gh pr create --title 'chore: Sync emoji data with GitHub emoji API' --body 'Found updated github emojis need to sync.' --base develop --reviewer docsifyjs/reviewers
+        env:
+          GH_TOKEN: ${{ github.token }}
```
Comment (sy-records) [inline .github/workflows/emoji.yml:41]:
```suggestion
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by sy-records** [comment]
good catch, The default token does not have `org:read` permissions and requires the use of PAT.

**Comment by Koooooo-7** [comment]
On the test here https://github.com/docsifyjs/docsify/actions/runs/9448830558/job/26023692530 
This action is updated to only works on a single branch `sync-emoji`, so we can not create multi PRs for it.
Problems: 
- We can not manually trigger the action before the branch related PR closed.
- It is not be a daily sync job since it can not run success before we merge the first PR if it exists. i.e.
   It run in the 2024-06-10 and raise a PR. If we merge it in 2024-06-15, the days between 06-10~06-15's jobs won't run success.
   And we are only able to check the changes in the sync day 2024-06-10. What the changes update after 2024-06-10, it only could be found on the next job run.

File: .github/workflows/emoji.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          branch: sync-emoji
+          create_branch: true
+          file_pattern: 'src/core/render/emoji-data.js docs/emoji.md'
+          push_options: '--force'
+
+      - name: Create Pull Request
+        if: ${{ steps.auto-commit-action.outputs.changes_detected == 'true' }}
+        run: |
+          gh pr create --title 'chore: Sync emoji data with GitHub emoji API' --body 'Found updated github emojis need to sync.' --base develop --reviewer docsifyjs/reviewers
+        env:
```
Comment (sy-records) [inline .github/workflows/emoji.yml:40]:
```suggestion
        continue-on-error: true
        env:
```

**Comment by Koooooo-7** [comment]
> There is no need for multiple branches and multiple PRs. The new commit will be forced to be pushed at runtime, so there is no need to create a new PR repeatedly.

I thought the behavior same to you mentioned that the opened PR should up to date, but the action is failed tho.

File: .github/workflows/emoji.yml (hunk 2 above)
Comment (Koooooo-7) [reply .github/workflows/emoji.yml:41]: > 

Looks good.

**Review by Koooooo-7 — APPROVED** [review]
LGTM.

**Pull request merged**